app/models/spree/order.rb

```diff
@@ -661,11 +661,12 @@ def after_cancel
       shipments.each(&:cancel!)
 
       OrderMailer.cancel_email(id).deliver_later
-      self.payment_state = 'credit_owed' unless shipped?
```

Comment:
I was a bit confused as to why this was added in the first place. Seems like in order to reflect the payment_state on the order, we would need to update the order via 'Update and Recalculate Fees' which in turn calls `update_payment_state`